tensorflow/core/kernels/mkl_qmatmul_op.cc

```
@@ -0,0 +1,875 @@
+          mkldnn_matmul_fwd_pd,
+      const Tensor& bias_tensor, const Tensor& weight_tensor) {
+    // If the bias is int32, it means the bias is already converted offline.
+    // and it can be added to matmul output directly.
+    if (std::is_same<Tbias, qint32>::value) {
+      return static_cast<Tbias*>(
+          const_cast<Tbias*>(bias_tensor.flat<Tbias>().data()));
+    } else {
+      // If the bias is fp32, then need to calculate the bias
+      const float min_input = context->input(3).flat<float>()(0);
+      const float max_input = context->input(4).flat<float>()(0);
+      const float min_weight = context->input(5).flat<float>()(0);
+      const float max_weight = context->input(6).flat<float>()(0);
+
+      std::vector<mkldnn::primitive> net;
+      float out_scale;
+      // If the bias is float and input quantize is MIN_FIRST bias has to be
+      // compensated with Bs32 = Qa * Qw * B𝑓32 + Qa * Min(A𝑓32) Ws8 * 1.
+      if (mode_ == QUANTIZE_MODE_MIN_FIRST) {
+        int k = weight_tensor.dim_size(0);
+        int n = weight_tensor.dim_size(1);
+        float* comp_bias = GetCompBiasBuffer(n);
+
+        qint8* wt_buf = static_cast<qint8*>(
+            const_cast<qint8*>(weight_tensor.flat<qint8>().data()));
+
+        const float* bias_buf = static_cast<float*>(
+            const_cast<float*>(bias_tensor.flat<float>().data()));
+
+        float qa_amin = 255 * min_input / (max_input - min_input);
+
+        out_scale = (255.0 * 127.0) /
+                    ((max_input - min_input) *
+                     std::max(std::abs(max_weight), std::abs(min_weight)));
+
+#pragma omp parallel for schedule(static)
+        for (int j = 0; j < n; ++j) {
+          int x = 0;
+          for (int i = 0; i < k; ++i) {
+            x += wt_buf[i * n + j];
+          }
+          comp_bias[j] =
+              ((bias_buf[j] * out_scale) + static_cast<float>(x * qa_amin));
+        }
+
+        return reinterpret_cast<Tbias*>(comp_bias_);
+
+      } else {
```

```suggestion
      } else if (mode_ == QUANTIZE_MODE_SCALE) {
```
Please also throw `errors::InvalidArgument` when `mode_` is unexpected.